compiler/steps/add-info.ts

```diff
@@ -19,18 +19,44 @@
 
 import * as model from '../model/metamodel'
 import { JsonSpec } from '../model/json-spec'
+import { execSync } from 'child_process'
+import { readFileSync } from 'fs'
+import { join } from 'path'
 
 /**
- * Adds the `_info` field to the JSON model
+ * Adds the `_info` field to the JSON model.
+ * If this code is being run in the base branch, it will update the version and hash
+ * fields with the value read from git, otherwise it will reuse the values
+ * currently stored in the output schema.
```

Comment:
I'm actually confused by `reuse the values currently stored in the output schema`: why don't we update the hash in that case?

It seems to me that the purpose of adding the hash to `schema.json` automatically after a change is to provide strong traceability information from downstream artefacts (e.g. generated code) to the spec that was used to produce them.

Am I missing something?

Reply:
We generate a new one if the generation starts from a base branch (main, 7.x, 7.n), while if you are generating the schema from a feature branch we don't care about the hash, so we reused the old one to have a nicer diff.

Reply:
Ah right, I was actually missing something and that makes sense!